File: app/src/main/java/net/sytes/kai_soft/letsbuyka/Application.java
package net.sytes.kai_soft.letsbuyka;

import android.content.Context;

/**
 * Created by Лунтя on 08.05.2018.
 */
/*  В этом классе мы открываем базу данных  */
public class Application extends android.app.Application {
    private static DataBase dataBase;
    private Context context;

    @Override
    public void onCreate() {
        dataBase = DataBase.getInstance(getApplicationContext());
        //DataBase.tableProducts.a1(dataBase);
        super.onCreate();
    }

    public static DataBase getDB(){
        return dataBase;
    }
}

File: app/src/main/java/net/sytes/kai_soft/letsbuyka/Application.kt
package net.sytes.kai_soft.letsbuyka

import android.util.Log
import com.google.firebase.auth.FirebaseAuth

class Application: android.app.Application() {
    companion object {
        private const val myTag = "letsbuy_Application"
        private lateinit var dataBase: DataBase
        private lateinit var context: Application
        private lateinit var mAuth: FirebaseAuth

        fun getDB(): DataBase{
            Log.i(myTag, "getDB()")
            return dataBase
        }
    }


    override fun onCreate() {
        Log.i(myTag, "onCreate()")
        context = this
        dataBase = DataBase.getInstance(context)
        super.onCreate()
    }
}